napari/utils/misc.py

```diff
@@ -390,3 +398,56 @@ def ensure_list_of_layer_data_tuple(val):
         except TypeError:
             pass
     raise TypeError('Not a valid list of layer data tuples!')
+
+
+def get_equality_operator(obj: Any) -> Callable[[Any, Any], bool]:
```

Comment:
Not needed in this PR, but I wonder if we'd start using this now in our EventedModel too. I know #2200 is still in progress, but might be nice to wire this up already.

Reply:
Yeah, I totally agree. As I think we discussed on zulip, it'd be nice if we can get away from `__equality_checks__` class attribute, and towards something that "just works" for the datatypes we use in napari.